### PR TITLE
Add "Cutting the Mustard" test

### PR DESCRIPTION
**Pull request by addyosmani**
We should define a CTM (Cutting the Mustard) test is, so WSK JavaScript components know what baseline functionality can safely be relied on. If a browser does not support our baseline, component handler logic defaults to noop.

This change will test for the presence of ClassList, querySelector, addEventListener and `Array.prototype.forEach`. In addition to upgrading components correctly, this change will add a `wsk-js` class to the document `<html>`. 

I've tested this out in evergreen browsers as well as IE9, 10 and 11. 

reviewers: @jasonmayes @sindresorhus @sgomes 

Prior art:

* http://responsivenews.co.uk/post/18948466399/cutting-the-mustard
* https://github.com/Financial-Times/ft-origami/issues/88
* http://codepen.io/atelierbram/pen/lepzJ

### CODE REVIEW

**Comment by addyosmani** [comment]
Typo removed :)

File: app/styleguide/wskComponentHandler.js [hunk 1]
```diff
@@ -151,5 +150,16 @@ var componentHandler = (function() {
...
   'use strict';
 
-  componentHandler.upgradeAllRegistered();
+  /**
+   * Performs a "Cutting the mustard" test. If the browser supports the features
+   * tested, adds a wsk-js class to the <html> element. It then upgrades all WSK
+   * components requiring JavaScript.
+   */
+  if ('classList' in document.createElement('div') && 'querySelector' in document && 'addEventListener' in window && Array.prototype.forEach) {
+    document.documentElement.className += 'wsk-js';
```
Comment (arthurvr) [inline app/styleguide/wskComponentHandler.js]: Why using `.className += 'wsk-js'` if the browser supports classList? This way of adding classes might result in trouble later:

screenshot 2015-01-26 20 24 18

Or at least add a space in the string you're appending, so it won't cause trouble.

**Comment by battaglr** [comment]
@addyosmani: hum, it's time for you to sign the CLA already. :P

**Comment by addyosmani** [comment]
I really need to fix the bot being overly aggressive. I'll get this updated to directly use classList shortly. Thanks for catching, Arthur!

File: app/styleguide/wskComponentHandler.js [hunk 2]
```diff
@@ -151,5 +150,16 @@ var componentHandler = (function() {
...
+    componentHandler.upgradeAllRegistered();
+  } else {
+    componentHandler.register = function () { /*noop*/ };
```
Comment (arthurvr) [inline app/styleguide/wskComponentHandler.js]: I wonder if having the `/*noop*/` comment in makes sense. I mean, an empty function is something we all know right?

File: app/styleguide/wskComponentHandler.js [hunk 3]
```diff
@@ -151,5 +150,16 @@ var componentHandler = (function() {
...
+    componentHandler.upgradeElement = function () { /*noop*/ };
```
Comment (arthurvr) [inline app/styleguide/wskComponentHandler.js]: It's all personal preference, but this seems cleaner to me:

```js
componentHandler.upgradeElement = componentHandler.register = function () { /*noop*/ };
```

File: app/styleguide/wskComponentHandler.js (hunk 2 above)
Comment (sindresorhus) [inline app/styleguide/wskComponentHandler.js]: noop is moot

Comment (addyosmani) [inline app/styleguide/wskComponentHandler.js]: Dropping in the next round of commits.

**Pull request merged**